src/features/Account/ManageAccountForm.tsx

```diff
@@ -0,0 +1,319 @@
+import React, { useState, useEffect } from 'react';
+import { useHistory } from 'react-router-dom';
+
+import {
+  ErrorMessage,
+  FastField,
+  Formik,
+  FormikProps,
+  FormikValues,
+} from 'formik';
+import { Account, Auth } from '../../api';
+import {
+  DietaryRestriction,
+  FrontendRoute,
+  Genders,
+  IAccount,
+  Pronouns,
+  UserType,
+} from '../../config';
+import * as CONSTANTS from '../../config/constants';
+import { Form, SubmitBtn } from '../../shared/Form';
+import * as FormikElements from '../../shared/Form/FormikElements';
+
+import ValidationErrorGenerator from '../../shared/Form/validationErrorGenerator';
+import WithToasterContainer from '../../shared/HOC/withToaster';
+import {
+  date2input,
+  getNestedAttr,
+  getOptionsFromEnum,
+  getValueFromQuery,
+  input2date,
+  isSponsor,
+} from '../../util';
+import getValidationSchema from './validationSchema';
+
+export enum ManageAccountModes {
+  CREATE,
+  EDIT,
+}
+
+interface IManageAccountProps {
+  mode: ManageAccountModes;
+}
+
+/**
+ * A form that lists all the account fields (username, password, email, etc)
+ * and either allows the user to create a new account (if in CREATE mode) or edit
+ * an existing account (if in EDIT mode)
+ */
+const ManageAccountForm: React.FC<IManageAccountProps> = (props) => {
+  // Get access to router history in order to programatically change page
+  const history = useHistory();
+
+  // Is the form submission currently being processed? (loading state)
+  const [isSubmitting, setIsSubmitting] = useState<boolean>(false);
+
+  // Track the details of the account that is either being created or updated
+  const [accountDetails, setAccountDetails] = useState<IAccount>({
+    accountType: (getValueFromQuery('accountType') as UserType) || UserType.UNKNOWN,
+    birthDate: '',
+    confirmed: false,
+    email: getNestedAttr(props, ['location', 'state', 'email']) || '',
+    firstName: '',
+    id: '',
+    lastName: '',
+    password: getNestedAttr(props, ['location', 'state', 'password']) || '',
+    phoneNumber: '',
+    pronoun: '',
+    gender: '',
+    dietaryRestrictions: [],
+  })
+
+  // If this is an edit form, load current account's data as default value
+  // for account details
+  useEffect(() => {
+    (async () => {
+      if (props.mode === ManageAccountModes.EDIT) {
+        try {
+          const response = await Account.getSelf();
+          const newAccountDetails = response.data.data;
+          newAccountDetails.birthDate = date2input(newAccountDetails.birthDate);
```

Comment:
What's the type of `newAccountDetails.birthDate`? Is it either an input or a date?

Reply:
It's a string before and after the date2input call

Reply:
Cool. What's the purpose of `date2input` again?

Reply:
Turns `"1998-11-30T04:00:00.000Z"` into `"11301998"`

Reply:
Cool. Resolving.